lacework/resource_lacework_report_rule.go

```diff
@@ -0,0 +1,576 @@
+		api.ReportRuleConfig{
+			Description:        d.Get("description").(string),
+			Severities:         severities,
+			ResourceGroups:     castStringSlice(resourceGroups),
+			EmailAlertChannels: castStringSlice(channels),
+			NotificationTypes:  notifications,
+		},
+	)
+
+	if err != nil {
+		return err
+	}
+
+	if !d.Get("enabled").(bool) {
+		reportRule.Filter.Enabled = 0
+	}
+
+	log.Printf("[INFO] Creating report rule with data:\n%+v\n", reportRule)
+	response, err := lacework.V2.ReportRules.Create(reportRule)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.Guid)
+	d.Set("name", response.Data.Filter.Name)
+	d.Set("guid", response.Data.Guid)
+	d.Set("enabled", response.Data.Filter.Enabled == 1)
+	d.Set("created_or_updated_time", response.Data.Filter.CreatedOrUpdatedTime)
+	d.Set("created_or_updated_by", response.Data.Filter.CreatedOrUpdatedBy)
+	d.Set("type", response.Data.Type)
+
+	log.Printf("[INFO] Created report rule with guid %s\n", response.Data.Guid)
+	return nil
+}
+
+func resourceLaceworkReportRuleRead(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework = meta.(*api.Client)
+		response api.ReportRuleResponse
+	)
+
+	log.Printf("[INFO] Reading report rule with guid %s\n", d.Id())
+	err := lacework.V2.ReportRules.Get(d.Id(), &response)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(response.Data.Guid)
+	d.Set("name", response.Data.Filter.Name)
+	d.Set("guid", response.Data.Guid)
+	d.Set("description", response.Data.Filter.Description)
+	d.Set("enabled", response.Data.Filter.Enabled == 1)
+	d.Set("created_or_updated_time", response.Data.Filter.CreatedOrUpdatedTime)
+	d.Set("created_or_updated_by", response.Data.Filter.CreatedOrUpdatedBy)
+	d.Set("type_name", response.Data.Type)
+	d.Set("severities", api.NewReportRuleSeveritiesFromIntSlice(response.Data.Filter.Severity).ToStringSlice())
+	d.Set("email_alert_channels", response.Data.EmailAlertChannels)
+	d.Set("resource_groups", response.Data.Filter.ResourceGroups)
+	setNotificationTypes(d, response.Data.ReportNotificationTypes)
+
+	log.Printf("[INFO] Read report rule with guid %s\n", response.Data.Guid)
+	return nil
+}
+
+func resourceLaceworkReportRuleUpdate(d *schema.ResourceData, meta interface{}) error {
+	var (
+		lacework       = meta.(*api.Client)
+		resourceGroups = d.Get("resource_groups").(*schema.Set).List()
+		severities     = api.NewReportRuleSeverities(castAttributeToStringSlice(d, "severities"))
+		channels       = d.Get("email_alert_channels").(*schema.Set).List()
+	)
+
+	notifications, err := getReportRuleNotifications(d)
+	if err != nil {
+		return err
+	}
+
+	reportRule, err := api.NewReportRule(d.Get("name").(string),
+		api.ReportRuleConfig{
+			Description:        d.Get("description").(string),
+			Severities:         severities,
+			ResourceGroups:     castStringSlice(resourceGroups),
+			EmailAlertChannels: castStringSlice(channels),
+			NotificationTypes:  notifications,
+		},
+	)
```

Comment:
Without the error check this could look much leaner like:
```suggestion
	reportRule, err := api.NewReportRule(d.Get("name").(string),
		api.ReportRuleConfig{
			Description:        d.Get("description").(string),
			Severities:         severities,
			ResourceGroups:     castStringSlice(resourceGroups),
			EmailAlertChannels: castStringSlice(channels),
			NotificationTypes:  getReportRuleNotifications(d),
		},
	)
```